src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1720,45 +1796,339 @@ class ActionDelegate {
         return menuItem
     }
 
-    protected Case doCreateMenuItem(String uri, String identifier, Case filter, Case orgGroup, Map<String, I18nString> allowedRoles, Map<String, I18nString> bannedRoles, List<String> defaultHeaders) {
-        if (findMenuItem(identifier)) {
-            throw new IllegalArgumentException("Menu item identifier $identifier is not unique!")
+    Case createMenuItem(MenuItemBody body) {
+        String sanitizedIdentifier = sanitize(body.identifier)
+
+        if (findMenuItem(sanitizedIdentifier)) {
```

Comment:
If the return value is ignored then it should be better to create `existsMenuItem` function that will only check if menu item with that identifier exists and returns true/false

Reply:
done

Reply:
What I had in mind was a new function in the `ElasticCaseService` that will check if the query matches any case in elasticsearch without instantiating java objects. Please check if it possible to use `exists` in Spring from this [SO answer](https://stackoverflow.com/a/45435650). If not it could be done using `count` operation.

Reply:
Reworked with `count` operation, since `exists` in Spring interface cannot be used with `query`